hack/tests/e2e-helm.sh

```diff
@@ -52,28 +47,6 @@ test_operator() {
         exit 1
     fi
 
-    metrics_service="nginx-operator-controller-manager-metrics-service"
-
-    # verify that metrics service was created
-    if ! timeout 60s bash -c -- "until kubectl get service/${metrics_service} --namespace=${operator_namespace} > /dev/null 2>&1; do sleep 1; done";
-    then
-        error_text "Failed to get metrics service"
-        operator_logs
-        exit 1
-    fi
-
-
-    serviceaccount_secret=$(kubectl get serviceaccounts default -n ${operator_namespace} -o jsonpath='{.secrets[0].name}')
-    token=$(kubectl get secret ${serviceaccount_secret} -n ${operator_namespace} -o jsonpath={.data.token} | base64 -d)
-
-    # verify that the metrics endpoint exists
-    if ! timeout 60s bash -c -- "until kubectl run --attach --rm --restart=Never --namespace=${operator_namespace} test-metrics --image=${metrics_test_image} -- -sfkH \"Authorization: Bearer ${token}\" https://${metrics_service}:8443/metrics; do sleep 1; done";
-    then
-        error_text "Failed to verify that metrics endpoint exists"
-        operator_logs
-        exit 1
-    fi
-
```

Comment:
All of this should be kept. We still have a metrics service in the Kubebuilder layout, but it is created alongside the operator as part of the kustomize manifests.

The only thing we need to remove from the e2e is the test for the presence of the `servicemonitors/nginx-operator-metrics` (which you've already done)